app/routes/task_routes.py

```diff
@@ -1 +1,173 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+from ..db import db
+from datetime import datetime, timezone
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+
+    #validate request data
+    if "title" not in request_body or "description" not in request_body:
+        response = {"details": "Invalid data"}
+        return make_response(response,400)
+
+    title = request_body["title"]
+    description = request_body["description"]
+    completed_at = request_body.get("completed_at")
+
+    new_task = Task(title=title, description=description, completed_at=completed_at)
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = {
+        "task":{
+            "id": new_task.id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": bool(new_task.completed_at)
+        }
+    }
+    return response, 201
+
+@tasks_bp.get('/<task_id>')
+def get_one_task(task_id):
+    task = validate_task(task_id)
+
+    return {
+        "task":{
+        "id": task.id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": bool(task.completed_at)
+        }
+    }
+
+@tasks_bp.get("")
+def get_all_tasks():
+
+    sort_param = request.args.get("sort", "").lower()
+    
+    if sort_param and sort_param.lower() == "asc":
+        query = db.select(Task).order_by(Task.title)
+
+    elif sort_param and sort_param.lower() == "desc":
+        query = db.select(Task).order_by(Task.title.desc())
+    else:
+        query = db.select(Task).order_by(Task.title)
+
+    tasks = db.session.scalars(query)
```

Comment:
Remember in class we wrote a function called `get_model_with_filters` though the implementation we had doesn't work in this project, the logic could be modified used here so we could DRY up our code when getting multiple records of a table. IT would also expand our functionality beyond what's required our the README.